{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e4d52995-8953-4aa1-9c88-b00682eb973c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7fb80c6a-81f8-4ebd-82bc-2ca951591644",
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b37d249a-8244-411d-b42e-a8344156a8e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_position</th>\n",
       "      <th>n_reads</th>\n",
       "      <th>probability_modified</th>\n",
       "      <th>kmer</th>\n",
       "      <th>mod_ratio</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>group_id</th>\n",
       "      <th>genome_pos</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dist_up_exon_junc</th>\n",
       "      <th>dist_down_exon_junc</th>\n",
       "      <th>region</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000263741</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1466</td>\n",
       "      <td>42</td>\n",
       "      <td>0.928761</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000354700</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>3222</td>\n",
       "      <td>78</td>\n",
       "      <td>0.955948</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.794872</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1300627.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>66.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1320</td>\n",
       "      <td>157</td>\n",
       "      <td>0.936434</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.687898</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1223856.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1753</td>\n",
       "      <td>194</td>\n",
       "      <td>0.938970</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.567010</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228538.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>409.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2612</td>\n",
       "      <td>31</td>\n",
       "      <td>0.952746</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.677419</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234848.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2612.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2579</td>\n",
       "      <td>27</td>\n",
       "      <td>0.995830</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.962963</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234815.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2579.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2432</td>\n",
       "      <td>22</td>\n",
       "      <td>0.987061</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234668.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2432.0</td>\n",
       "      <td>372.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000416718</td>\n",
       "      <td>ENSG00000198744</td>\n",
       "      <td>341</td>\n",
       "      <td>801</td>\n",
       "      <td>0.927230</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>634716.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>341.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>MTCO3P12</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENST00000465727</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1482</td>\n",
       "      <td>42</td>\n",
       "      <td>0.949604</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENST00000467278</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>2633</td>\n",
       "      <td>76</td>\n",
       "      <td>0.966780</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.868421</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1298659.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>297.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1981</td>\n",
       "      <td>68</td>\n",
       "      <td>0.936208</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.779412</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219068.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1226.0</td>\n",
       "      <td>609.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2414</td>\n",
       "      <td>97</td>\n",
       "      <td>0.911677</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.556701</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218635.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1659.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENST00000492936</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>4747</td>\n",
       "      <td>61</td>\n",
       "      <td>0.956273</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.770492</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1304658.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1272.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2904</td>\n",
       "      <td>81</td>\n",
       "      <td>0.941420</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.765432</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219381.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>611.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>3337</td>\n",
       "      <td>111</td>\n",
       "      <td>0.903208</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.558559</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218948.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2404.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6171</td>\n",
       "      <td>23</td>\n",
       "      <td>0.965541</td>\n",
       "      <td>AGACT</td>\n",
       "      <td>0.913043</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>778607.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>82.0</td>\n",
       "      <td>322.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6083</td>\n",
       "      <td>29</td>\n",
       "      <td>0.980850</td>\n",
       "      <td>GAACT</td>\n",
       "      <td>0.655172</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>774177.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>104.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      transcript_id          gene_id  transcript_position  n_reads  \\\n",
       "0   ENST00000263741  ENSG00000078808                 1466       42   \n",
       "1   ENST00000354700  ENSG00000131584                 3222       78   \n",
       "2   ENST00000360001  ENSG00000078808                 1320      157   \n",
       "3   ENST00000360001  ENSG00000078808                 1753      194   \n",
       "4   ENST00000379198  ENSG00000176022                 2612       31   \n",
       "5   ENST00000379198  ENSG00000176022                 2579       27   \n",
       "6   ENST00000379198  ENSG00000176022                 2432       22   \n",
       "7   ENST00000416718  ENSG00000198744                  341      801   \n",
       "8   ENST00000465727  ENSG00000078808                 1482       42   \n",
       "9   ENST00000467278  ENSG00000131584                 2633       76   \n",
       "10  ENST00000478938  ENSG00000078808                 1981       68   \n",
       "11  ENST00000478938  ENSG00000078808                 2414       97   \n",
       "12  ENST00000492936  ENSG00000131584                 4747       61   \n",
       "13  ENST00000494748  ENSG00000078808                 2904       81   \n",
       "14  ENST00000494748  ENSG00000078808                 3337      111   \n",
       "15  ENST00000506640  ENSG00000228327                 6171       23   \n",
       "16  ENST00000506640  ENSG00000228327                 6083       29   \n",
       "\n",
       "    probability_modified   kmer  mod_ratio sample_id group_id  genome_pos  \\\n",
       "0               0.928761  GGACT   0.761905        s2     caud   1228941.0   \n",
       "1               0.955948  AGACA   0.794872        s2     caud   1300627.0   \n",
       "2               0.936434  GGACT   0.687898        s1     caud   1223856.0   \n",
       "3               0.938970  AAACT   0.567010        s1     caud   1228538.0   \n",
       "4               0.952746  GGACT   0.677419        s2     caud   1234848.0   \n",
       "5               0.995830  GGACA   0.962963        s2     caud   1234815.0   \n",
       "6               0.987061  GGACT   0.909091        s2     caud   1234668.0   \n",
       "7               0.927230  GGACA   0.666667        s1     caud    634716.0   \n",
       "8               0.949604  GGACT   0.761905        s2     caud   1228941.0   \n",
       "9               0.966780  AGACA   0.868421        s2     caud   1298659.0   \n",
       "10              0.936208  GGACT   0.779412        s2     caud   1219068.0   \n",
       "11              0.911677  AAACT   0.556701        s2     caud   1218635.0   \n",
       "12              0.956273  AGACA   0.770492        s2     caud   1304658.0   \n",
       "13              0.941420  GGACT   0.765432        s2     caud   1219381.0   \n",
       "14              0.903208  AAACT   0.558559        s2     caud   1218948.0   \n",
       "15              0.965541  AGACT   0.913043        s1     caud    778607.0   \n",
       "16              0.980850  GAACT   0.655172        s1     caud    774177.0   \n",
       "\n",
       "   chromosome  dist_up_exon_junc  dist_down_exon_junc region  gene_name  \\\n",
       "0        chr1                6.0                472.0   UTR3       SDF4   \n",
       "1        chr1               66.0                117.0   UTR3      ACAP3   \n",
       "2        chr1              113.0                 23.0    CDS       SDF4   \n",
       "3        chr1              409.0                 69.0   UTR3       SDF4   \n",
       "4        chr1             2612.0                192.0   UTR3    B3GALT6   \n",
       "5        chr1             2579.0                225.0   UTR3    B3GALT6   \n",
       "6        chr1             2432.0                372.0   UTR3    B3GALT6   \n",
       "7        chr1              341.0                205.0  ncRNA   MTCO3P12   \n",
       "8        chr1                6.0                472.0   UTR3       SDF4   \n",
       "9        chr1              297.0                 91.0  ncRNA      ACAP3   \n",
       "10       chr1             1226.0                609.0  ncRNA       SDF4   \n",
       "11       chr1             1659.0                176.0  ncRNA       SDF4   \n",
       "12       chr1             1272.0                571.0  ncRNA      ACAP3   \n",
       "13       chr1             1971.0                611.0  ncRNA       SDF4   \n",
       "14       chr1             2404.0                178.0  ncRNA       SDF4   \n",
       "15       chr1               82.0                322.0  ncRNA  LINC00115   \n",
       "16       chr1              104.0                  5.0  ncRNA  LINC00115   \n",
       "\n",
       "            transcript_type  \n",
       "0            protein_coding  \n",
       "1            protein_coding  \n",
       "2            protein_coding  \n",
       "3            protein_coding  \n",
       "4            protein_coding  \n",
       "5            protein_coding  \n",
       "6            protein_coding  \n",
       "7    unprocessed_pseudogene  \n",
       "8   nonsense_mediated_decay  \n",
       "9           retained_intron  \n",
       "10          retained_intron  \n",
       "11          retained_intron  \n",
       "12          retained_intron  \n",
       "13          retained_intron  \n",
       "14          retained_intron  \n",
       "15                   lncRNA  \n",
       "16                   lncRNA  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m6anet = pd.read_csv(\"output_perfix_test_data_m6anet_mod_sites.csv\")\n",
    "m6anet = m6anet.dropna(subset=['genome_pos', 'chromosome'])\n",
    "m6anet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7a228fa4-075a-46c6-bec3-460f25f98001",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_position</th>\n",
       "      <th>n_reads</th>\n",
       "      <th>probability_modified</th>\n",
       "      <th>kmer</th>\n",
       "      <th>mod_ratio</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>group_id</th>\n",
       "      <th>genome_pos</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dist_up_exon_junc</th>\n",
       "      <th>dist_down_exon_junc</th>\n",
       "      <th>region</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000263741</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1466</td>\n",
       "      <td>42</td>\n",
       "      <td>0.928761</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000354700</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>3222</td>\n",
       "      <td>78</td>\n",
       "      <td>0.955948</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.794872</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1300627.0</td>\n",
       "      <td>1</td>\n",
       "      <td>66.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1320</td>\n",
       "      <td>157</td>\n",
       "      <td>0.936434</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.687898</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1223856.0</td>\n",
       "      <td>1</td>\n",
       "      <td>113.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1753</td>\n",
       "      <td>194</td>\n",
       "      <td>0.938970</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.567010</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228538.0</td>\n",
       "      <td>1</td>\n",
       "      <td>409.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2612</td>\n",
       "      <td>31</td>\n",
       "      <td>0.952746</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.677419</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234848.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2612.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2579</td>\n",
       "      <td>27</td>\n",
       "      <td>0.995830</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.962963</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234815.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2579.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2432</td>\n",
       "      <td>22</td>\n",
       "      <td>0.987061</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234668.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2432.0</td>\n",
       "      <td>372.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000416718</td>\n",
       "      <td>ENSG00000198744</td>\n",
       "      <td>341</td>\n",
       "      <td>801</td>\n",
       "      <td>0.927230</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>634716.0</td>\n",
       "      <td>1</td>\n",
       "      <td>341.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>MTCO3P12</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENST00000465727</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1482</td>\n",
       "      <td>42</td>\n",
       "      <td>0.949604</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENST00000467278</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>2633</td>\n",
       "      <td>76</td>\n",
       "      <td>0.966780</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.868421</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1298659.0</td>\n",
       "      <td>1</td>\n",
       "      <td>297.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1981</td>\n",
       "      <td>68</td>\n",
       "      <td>0.936208</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.779412</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219068.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1226.0</td>\n",
       "      <td>609.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2414</td>\n",
       "      <td>97</td>\n",
       "      <td>0.911677</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.556701</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218635.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1659.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENST00000492936</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>4747</td>\n",
       "      <td>61</td>\n",
       "      <td>0.956273</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.770492</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1304658.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1272.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2904</td>\n",
       "      <td>81</td>\n",
       "      <td>0.941420</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.765432</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219381.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>611.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>3337</td>\n",
       "      <td>111</td>\n",
       "      <td>0.903208</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.558559</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218948.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2404.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6171</td>\n",
       "      <td>23</td>\n",
       "      <td>0.965541</td>\n",
       "      <td>AGACT</td>\n",
       "      <td>0.913043</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>778607.0</td>\n",
       "      <td>1</td>\n",
       "      <td>82.0</td>\n",
       "      <td>322.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6083</td>\n",
       "      <td>29</td>\n",
       "      <td>0.980850</td>\n",
       "      <td>GAACT</td>\n",
       "      <td>0.655172</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>774177.0</td>\n",
       "      <td>1</td>\n",
       "      <td>104.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      transcript_id          gene_id  transcript_position  n_reads  \\\n",
       "0   ENST00000263741  ENSG00000078808                 1466       42   \n",
       "1   ENST00000354700  ENSG00000131584                 3222       78   \n",
       "2   ENST00000360001  ENSG00000078808                 1320      157   \n",
       "3   ENST00000360001  ENSG00000078808                 1753      194   \n",
       "4   ENST00000379198  ENSG00000176022                 2612       31   \n",
       "5   ENST00000379198  ENSG00000176022                 2579       27   \n",
       "6   ENST00000379198  ENSG00000176022                 2432       22   \n",
       "7   ENST00000416718  ENSG00000198744                  341      801   \n",
       "8   ENST00000465727  ENSG00000078808                 1482       42   \n",
       "9   ENST00000467278  ENSG00000131584                 2633       76   \n",
       "10  ENST00000478938  ENSG00000078808                 1981       68   \n",
       "11  ENST00000478938  ENSG00000078808                 2414       97   \n",
       "12  ENST00000492936  ENSG00000131584                 4747       61   \n",
       "13  ENST00000494748  ENSG00000078808                 2904       81   \n",
       "14  ENST00000494748  ENSG00000078808                 3337      111   \n",
       "15  ENST00000506640  ENSG00000228327                 6171       23   \n",
       "16  ENST00000506640  ENSG00000228327                 6083       29   \n",
       "\n",
       "    probability_modified   kmer  mod_ratio sample_id group_id  genome_pos  \\\n",
       "0               0.928761  GGACT   0.761905        s2     caud   1228941.0   \n",
       "1               0.955948  AGACA   0.794872        s2     caud   1300627.0   \n",
       "2               0.936434  GGACT   0.687898        s1     caud   1223856.0   \n",
       "3               0.938970  AAACT   0.567010        s1     caud   1228538.0   \n",
       "4               0.952746  GGACT   0.677419        s2     caud   1234848.0   \n",
       "5               0.995830  GGACA   0.962963        s2     caud   1234815.0   \n",
       "6               0.987061  GGACT   0.909091        s2     caud   1234668.0   \n",
       "7               0.927230  GGACA   0.666667        s1     caud    634716.0   \n",
       "8               0.949604  GGACT   0.761905        s2     caud   1228941.0   \n",
       "9               0.966780  AGACA   0.868421        s2     caud   1298659.0   \n",
       "10              0.936208  GGACT   0.779412        s2     caud   1219068.0   \n",
       "11              0.911677  AAACT   0.556701        s2     caud   1218635.0   \n",
       "12              0.956273  AGACA   0.770492        s2     caud   1304658.0   \n",
       "13              0.941420  GGACT   0.765432        s2     caud   1219381.0   \n",
       "14              0.903208  AAACT   0.558559        s2     caud   1218948.0   \n",
       "15              0.965541  AGACT   0.913043        s1     caud    778607.0   \n",
       "16              0.980850  GAACT   0.655172        s1     caud    774177.0   \n",
       "\n",
       "   chromosome  dist_up_exon_junc  dist_down_exon_junc region  gene_name  \\\n",
       "0           1                6.0                472.0   UTR3       SDF4   \n",
       "1           1               66.0                117.0   UTR3      ACAP3   \n",
       "2           1              113.0                 23.0    CDS       SDF4   \n",
       "3           1              409.0                 69.0   UTR3       SDF4   \n",
       "4           1             2612.0                192.0   UTR3    B3GALT6   \n",
       "5           1             2579.0                225.0   UTR3    B3GALT6   \n",
       "6           1             2432.0                372.0   UTR3    B3GALT6   \n",
       "7           1              341.0                205.0  ncRNA   MTCO3P12   \n",
       "8           1                6.0                472.0   UTR3       SDF4   \n",
       "9           1              297.0                 91.0  ncRNA      ACAP3   \n",
       "10          1             1226.0                609.0  ncRNA       SDF4   \n",
       "11          1             1659.0                176.0  ncRNA       SDF4   \n",
       "12          1             1272.0                571.0  ncRNA      ACAP3   \n",
       "13          1             1971.0                611.0  ncRNA       SDF4   \n",
       "14          1             2404.0                178.0  ncRNA       SDF4   \n",
       "15          1               82.0                322.0  ncRNA  LINC00115   \n",
       "16          1              104.0                  5.0  ncRNA  LINC00115   \n",
       "\n",
       "            transcript_type  \n",
       "0            protein_coding  \n",
       "1            protein_coding  \n",
       "2            protein_coding  \n",
       "3            protein_coding  \n",
       "4            protein_coding  \n",
       "5            protein_coding  \n",
       "6            protein_coding  \n",
       "7    unprocessed_pseudogene  \n",
       "8   nonsense_mediated_decay  \n",
       "9           retained_intron  \n",
       "10          retained_intron  \n",
       "11          retained_intron  \n",
       "12          retained_intron  \n",
       "13          retained_intron  \n",
       "14          retained_intron  \n",
       "15                   lncRNA  \n",
       "16                   lncRNA  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Normalize chromosome format\n",
    "m6anet['chromosome'] = m6anet['chromosome'].astype(str).str.lower().str.replace('chr', '')\n",
    "m6anet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "790b39a8-7f20-48df-bb78-d88e28f8ea0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_position</th>\n",
       "      <th>n_reads</th>\n",
       "      <th>probability_modified</th>\n",
       "      <th>kmer</th>\n",
       "      <th>mod_ratio</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>group_id</th>\n",
       "      <th>genome_pos</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dist_up_exon_junc</th>\n",
       "      <th>dist_down_exon_junc</th>\n",
       "      <th>region</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000263741</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1466</td>\n",
       "      <td>42</td>\n",
       "      <td>0.928761</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941</td>\n",
       "      <td>1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000354700</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>3222</td>\n",
       "      <td>78</td>\n",
       "      <td>0.955948</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.794872</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1300627</td>\n",
       "      <td>1</td>\n",
       "      <td>66.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1320</td>\n",
       "      <td>157</td>\n",
       "      <td>0.936434</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.687898</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1223856</td>\n",
       "      <td>1</td>\n",
       "      <td>113.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000360001</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1753</td>\n",
       "      <td>194</td>\n",
       "      <td>0.938970</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.567010</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228538</td>\n",
       "      <td>1</td>\n",
       "      <td>409.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2612</td>\n",
       "      <td>31</td>\n",
       "      <td>0.952746</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.677419</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234848</td>\n",
       "      <td>1</td>\n",
       "      <td>2612.0</td>\n",
       "      <td>192.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2579</td>\n",
       "      <td>27</td>\n",
       "      <td>0.995830</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.962963</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234815</td>\n",
       "      <td>1</td>\n",
       "      <td>2579.0</td>\n",
       "      <td>225.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>ENSG00000176022</td>\n",
       "      <td>2432</td>\n",
       "      <td>22</td>\n",
       "      <td>0.987061</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1234668</td>\n",
       "      <td>1</td>\n",
       "      <td>2432.0</td>\n",
       "      <td>372.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000416718</td>\n",
       "      <td>ENSG00000198744</td>\n",
       "      <td>341</td>\n",
       "      <td>801</td>\n",
       "      <td>0.927230</td>\n",
       "      <td>GGACA</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>634716</td>\n",
       "      <td>1</td>\n",
       "      <td>341.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>MTCO3P12</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENST00000465727</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1482</td>\n",
       "      <td>42</td>\n",
       "      <td>0.949604</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.761905</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1228941</td>\n",
       "      <td>1</td>\n",
       "      <td>6.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENST00000467278</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>2633</td>\n",
       "      <td>76</td>\n",
       "      <td>0.966780</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.868421</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1298659</td>\n",
       "      <td>1</td>\n",
       "      <td>297.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>1981</td>\n",
       "      <td>68</td>\n",
       "      <td>0.936208</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.779412</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219068</td>\n",
       "      <td>1</td>\n",
       "      <td>1226.0</td>\n",
       "      <td>609.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000478938</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2414</td>\n",
       "      <td>97</td>\n",
       "      <td>0.911677</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.556701</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218635</td>\n",
       "      <td>1</td>\n",
       "      <td>1659.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENST00000492936</td>\n",
       "      <td>ENSG00000131584</td>\n",
       "      <td>4747</td>\n",
       "      <td>61</td>\n",
       "      <td>0.956273</td>\n",
       "      <td>AGACA</td>\n",
       "      <td>0.770492</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1304658</td>\n",
       "      <td>1</td>\n",
       "      <td>1272.0</td>\n",
       "      <td>571.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>2904</td>\n",
       "      <td>81</td>\n",
       "      <td>0.941420</td>\n",
       "      <td>GGACT</td>\n",
       "      <td>0.765432</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1219381</td>\n",
       "      <td>1</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>611.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000494748</td>\n",
       "      <td>ENSG00000078808</td>\n",
       "      <td>3337</td>\n",
       "      <td>111</td>\n",
       "      <td>0.903208</td>\n",
       "      <td>AAACT</td>\n",
       "      <td>0.558559</td>\n",
       "      <td>s2</td>\n",
       "      <td>caud</td>\n",
       "      <td>1218948</td>\n",
       "      <td>1</td>\n",
       "      <td>2404.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>retained_intron</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6171</td>\n",
       "      <td>23</td>\n",
       "      <td>0.965541</td>\n",
       "      <td>AGACT</td>\n",
       "      <td>0.913043</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>778607</td>\n",
       "      <td>1</td>\n",
       "      <td>82.0</td>\n",
       "      <td>322.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENST00000506640</td>\n",
       "      <td>ENSG00000228327</td>\n",
       "      <td>6083</td>\n",
       "      <td>29</td>\n",
       "      <td>0.980850</td>\n",
       "      <td>GAACT</td>\n",
       "      <td>0.655172</td>\n",
       "      <td>s1</td>\n",
       "      <td>caud</td>\n",
       "      <td>774177</td>\n",
       "      <td>1</td>\n",
       "      <td>104.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>LINC00115</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      transcript_id          gene_id  transcript_position  n_reads  \\\n",
       "0   ENST00000263741  ENSG00000078808                 1466       42   \n",
       "1   ENST00000354700  ENSG00000131584                 3222       78   \n",
       "2   ENST00000360001  ENSG00000078808                 1320      157   \n",
       "3   ENST00000360001  ENSG00000078808                 1753      194   \n",
       "4   ENST00000379198  ENSG00000176022                 2612       31   \n",
       "5   ENST00000379198  ENSG00000176022                 2579       27   \n",
       "6   ENST00000379198  ENSG00000176022                 2432       22   \n",
       "7   ENST00000416718  ENSG00000198744                  341      801   \n",
       "8   ENST00000465727  ENSG00000078808                 1482       42   \n",
       "9   ENST00000467278  ENSG00000131584                 2633       76   \n",
       "10  ENST00000478938  ENSG00000078808                 1981       68   \n",
       "11  ENST00000478938  ENSG00000078808                 2414       97   \n",
       "12  ENST00000492936  ENSG00000131584                 4747       61   \n",
       "13  ENST00000494748  ENSG00000078808                 2904       81   \n",
       "14  ENST00000494748  ENSG00000078808                 3337      111   \n",
       "15  ENST00000506640  ENSG00000228327                 6171       23   \n",
       "16  ENST00000506640  ENSG00000228327                 6083       29   \n",
       "\n",
       "    probability_modified   kmer  mod_ratio sample_id group_id  genome_pos  \\\n",
       "0               0.928761  GGACT   0.761905        s2     caud     1228941   \n",
       "1               0.955948  AGACA   0.794872        s2     caud     1300627   \n",
       "2               0.936434  GGACT   0.687898        s1     caud     1223856   \n",
       "3               0.938970  AAACT   0.567010        s1     caud     1228538   \n",
       "4               0.952746  GGACT   0.677419        s2     caud     1234848   \n",
       "5               0.995830  GGACA   0.962963        s2     caud     1234815   \n",
       "6               0.987061  GGACT   0.909091        s2     caud     1234668   \n",
       "7               0.927230  GGACA   0.666667        s1     caud      634716   \n",
       "8               0.949604  GGACT   0.761905        s2     caud     1228941   \n",
       "9               0.966780  AGACA   0.868421        s2     caud     1298659   \n",
       "10              0.936208  GGACT   0.779412        s2     caud     1219068   \n",
       "11              0.911677  AAACT   0.556701        s2     caud     1218635   \n",
       "12              0.956273  AGACA   0.770492        s2     caud     1304658   \n",
       "13              0.941420  GGACT   0.765432        s2     caud     1219381   \n",
       "14              0.903208  AAACT   0.558559        s2     caud     1218948   \n",
       "15              0.965541  AGACT   0.913043        s1     caud      778607   \n",
       "16              0.980850  GAACT   0.655172        s1     caud      774177   \n",
       "\n",
       "   chromosome  dist_up_exon_junc  dist_down_exon_junc region  gene_name  \\\n",
       "0           1                6.0                472.0   UTR3       SDF4   \n",
       "1           1               66.0                117.0   UTR3      ACAP3   \n",
       "2           1              113.0                 23.0    CDS       SDF4   \n",
       "3           1              409.0                 69.0   UTR3       SDF4   \n",
       "4           1             2612.0                192.0   UTR3    B3GALT6   \n",
       "5           1             2579.0                225.0   UTR3    B3GALT6   \n",
       "6           1             2432.0                372.0   UTR3    B3GALT6   \n",
       "7           1              341.0                205.0  ncRNA   MTCO3P12   \n",
       "8           1                6.0                472.0   UTR3       SDF4   \n",
       "9           1              297.0                 91.0  ncRNA      ACAP3   \n",
       "10          1             1226.0                609.0  ncRNA       SDF4   \n",
       "11          1             1659.0                176.0  ncRNA       SDF4   \n",
       "12          1             1272.0                571.0  ncRNA      ACAP3   \n",
       "13          1             1971.0                611.0  ncRNA       SDF4   \n",
       "14          1             2404.0                178.0  ncRNA       SDF4   \n",
       "15          1               82.0                322.0  ncRNA  LINC00115   \n",
       "16          1              104.0                  5.0  ncRNA  LINC00115   \n",
       "\n",
       "            transcript_type  \n",
       "0            protein_coding  \n",
       "1            protein_coding  \n",
       "2            protein_coding  \n",
       "3            protein_coding  \n",
       "4            protein_coding  \n",
       "5            protein_coding  \n",
       "6            protein_coding  \n",
       "7    unprocessed_pseudogene  \n",
       "8   nonsense_mediated_decay  \n",
       "9           retained_intron  \n",
       "10          retained_intron  \n",
       "11          retained_intron  \n",
       "12          retained_intron  \n",
       "13          retained_intron  \n",
       "14          retained_intron  \n",
       "15                   lncRNA  \n",
       "16                   lncRNA  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert genome_pos to integer\n",
    "m6anet['genome_pos'] = m6anet['genome_pos'].astype(int)\n",
    "m6anet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3347d061-e30c-476d-8075-2c4aee76038a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_m6add_files(datatype):\n",
    "    files = glob.glob(\"GSE*_\"+datatype+\".txt\")\n",
    "    m6add = []\n",
    "    for file in RADAR_files:\n",
    "        df = pd.read_csv(file, sep = \"\\t\")\n",
    "        m6add.append(df)\n",
    "    return pd.concat(m6add)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9cdd00fe-b412-49fb-b7d1-50da5bccf2bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Disease</th>\n",
       "      <th>Symbol_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>p_value</th>\n",
       "      <th>logFC</th>\n",
       "      <th>GSE</th>\n",
       "      <th>Treat_method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>NLRP7</td>\n",
       "      <td>chr19</td>\n",
       "      <td>-</td>\n",
       "      <td>54939852</td>\n",
       "      <td>54939901</td>\n",
       "      <td>9.376180e-03</td>\n",
       "      <td>-3.135494</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>RP11-626G11.6</td>\n",
       "      <td>chr16</td>\n",
       "      <td>-</td>\n",
       "      <td>19111185</td>\n",
       "      <td>19111234</td>\n",
       "      <td>1.473765e-02</td>\n",
       "      <td>-2.251292</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>chrY</td>\n",
       "      <td>+</td>\n",
       "      <td>56963000</td>\n",
       "      <td>56963049</td>\n",
       "      <td>1.740151e-02</td>\n",
       "      <td>-2.140066</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>KBTBD3</td>\n",
       "      <td>chr11</td>\n",
       "      <td>-</td>\n",
       "      <td>106053831</td>\n",
       "      <td>106053880</td>\n",
       "      <td>4.911331e-03</td>\n",
       "      <td>-1.853891</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>RP11-113D6.10</td>\n",
       "      <td>chr11</td>\n",
       "      <td>+</td>\n",
       "      <td>18210349</td>\n",
       "      <td>18210398</td>\n",
       "      <td>2.568635e-03</td>\n",
       "      <td>-1.739703</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>RP11-1212A22.4</td>\n",
       "      <td>chr16</td>\n",
       "      <td>-</td>\n",
       "      <td>18379183</td>\n",
       "      <td>18379232</td>\n",
       "      <td>5.820000e-09</td>\n",
       "      <td>3.157000</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>PKD1P6</td>\n",
       "      <td>chr16</td>\n",
       "      <td>-</td>\n",
       "      <td>15136874</td>\n",
       "      <td>15136923</td>\n",
       "      <td>1.400000e-09</td>\n",
       "      <td>3.020425</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>PKD1P1</td>\n",
       "      <td>chr16</td>\n",
       "      <td>+</td>\n",
       "      <td>16318662</td>\n",
       "      <td>16318711</td>\n",
       "      <td>2.370000e-07</td>\n",
       "      <td>2.890372</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>+</td>\n",
       "      <td>50416152</td>\n",
       "      <td>50416201</td>\n",
       "      <td>1.100000e-07</td>\n",
       "      <td>2.708050</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>STRA6</td>\n",
       "      <td>chr15</td>\n",
       "      <td>-</td>\n",
       "      <td>74194956</td>\n",
       "      <td>74195005</td>\n",
       "      <td>7.120000e-06</td>\n",
       "      <td>2.442347</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>226341 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Disease       Symbol_id    chr strand      start        end  \\\n",
       "0   Ovarian cancers           NLRP7  chr19      -   54939852   54939901   \n",
       "1   Ovarian cancers   RP11-626G11.6  chr16      -   19111185   19111234   \n",
       "2   Ovarian cancers           SPRY3   chrY      +   56963000   56963049   \n",
       "3   Ovarian cancers          KBTBD3  chr11      -  106053831  106053880   \n",
       "4   Ovarian cancers   RP11-113D6.10  chr11      +   18210349   18210398   \n",
       "..              ...             ...    ...    ...        ...        ...   \n",
       "5      Liver cancer  RP11-1212A22.4  chr16      -   18379183   18379232   \n",
       "6      Liver cancer          PKD1P6  chr16      -   15136874   15136923   \n",
       "7      Liver cancer          PKD1P1  chr16      +   16318662   16318711   \n",
       "8      Liver cancer           POLD1  chr19      +   50416152   50416201   \n",
       "9      Liver cancer           STRA6  chr15      -   74194956   74195005   \n",
       "\n",
       "         p_value     logFC        GSE      Treat_method  \n",
       "0   9.376180e-03 -3.135494  GSE119963   PARPi resistant  \n",
       "1   1.473765e-02 -2.251292  GSE119963   PARPi resistant  \n",
       "2   1.740151e-02 -2.140066  GSE119963   PARPi resistant  \n",
       "3   4.911331e-03 -1.853891  GSE119963   PARPi resistant  \n",
       "4   2.568635e-03 -1.739703  GSE119963   PARPi resistant  \n",
       "..           ...       ...        ...               ...  \n",
       "5   5.820000e-09  3.157000  GSE102620  METTL3 knockdown  \n",
       "6   1.400000e-09  3.020425  GSE102620  METTL3 knockdown  \n",
       "7   2.370000e-07  2.890372  GSE102620  METTL3 knockdown  \n",
       "8   1.100000e-07  2.708050  GSE102620  METTL3 knockdown  \n",
       "9   7.120000e-06  2.442347  GSE102620  METTL3 knockdown  \n",
       "\n",
       "[226341 rows x 10 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datatype = \"MeTDiff\"\n",
    "m6add_df = get_m6add_files(datatype)\n",
    "m6add_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a43474a1-0d28-4166-95dd-26046994ff62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Disease</th>\n",
       "      <th>Symbol_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>p_value</th>\n",
       "      <th>logFC</th>\n",
       "      <th>GSE</th>\n",
       "      <th>Treat_method</th>\n",
       "      <th>genome_pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>NLRP7</td>\n",
       "      <td>19</td>\n",
       "      <td>-</td>\n",
       "      <td>54939852</td>\n",
       "      <td>54939901</td>\n",
       "      <td>9.376180e-03</td>\n",
       "      <td>-3.135494</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "      <td>54939876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>RP11-626G11.6</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>19111185</td>\n",
       "      <td>19111234</td>\n",
       "      <td>1.473765e-02</td>\n",
       "      <td>-2.251292</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "      <td>19111209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>y</td>\n",
       "      <td>+</td>\n",
       "      <td>56963000</td>\n",
       "      <td>56963049</td>\n",
       "      <td>1.740151e-02</td>\n",
       "      <td>-2.140066</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "      <td>56963024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>KBTBD3</td>\n",
       "      <td>11</td>\n",
       "      <td>-</td>\n",
       "      <td>106053831</td>\n",
       "      <td>106053880</td>\n",
       "      <td>4.911331e-03</td>\n",
       "      <td>-1.853891</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "      <td>106053855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ovarian cancers</td>\n",
       "      <td>RP11-113D6.10</td>\n",
       "      <td>11</td>\n",
       "      <td>+</td>\n",
       "      <td>18210349</td>\n",
       "      <td>18210398</td>\n",
       "      <td>2.568635e-03</td>\n",
       "      <td>-1.739703</td>\n",
       "      <td>GSE119963</td>\n",
       "      <td>PARPi resistant</td>\n",
       "      <td>18210373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>RP11-1212A22.4</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>18379183</td>\n",
       "      <td>18379232</td>\n",
       "      <td>5.820000e-09</td>\n",
       "      <td>3.157000</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "      <td>18379207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>PKD1P6</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>15136874</td>\n",
       "      <td>15136923</td>\n",
       "      <td>1.400000e-09</td>\n",
       "      <td>3.020425</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "      <td>15136898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>PKD1P1</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>16318662</td>\n",
       "      <td>16318711</td>\n",
       "      <td>2.370000e-07</td>\n",
       "      <td>2.890372</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "      <td>16318686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>19</td>\n",
       "      <td>+</td>\n",
       "      <td>50416152</td>\n",
       "      <td>50416201</td>\n",
       "      <td>1.100000e-07</td>\n",
       "      <td>2.708050</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "      <td>50416176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Liver cancer</td>\n",
       "      <td>STRA6</td>\n",
       "      <td>15</td>\n",
       "      <td>-</td>\n",
       "      <td>74194956</td>\n",
       "      <td>74195005</td>\n",
       "      <td>7.120000e-06</td>\n",
       "      <td>2.442347</td>\n",
       "      <td>GSE102620</td>\n",
       "      <td>METTL3 knockdown</td>\n",
       "      <td>74194980</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>226341 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Disease       Symbol_id chr strand      start        end  \\\n",
       "0   Ovarian cancers           NLRP7  19      -   54939852   54939901   \n",
       "1   Ovarian cancers   RP11-626G11.6  16      -   19111185   19111234   \n",
       "2   Ovarian cancers           SPRY3   y      +   56963000   56963049   \n",
       "3   Ovarian cancers          KBTBD3  11      -  106053831  106053880   \n",
       "4   Ovarian cancers   RP11-113D6.10  11      +   18210349   18210398   \n",
       "..              ...             ...  ..    ...        ...        ...   \n",
       "5      Liver cancer  RP11-1212A22.4  16      -   18379183   18379232   \n",
       "6      Liver cancer          PKD1P6  16      -   15136874   15136923   \n",
       "7      Liver cancer          PKD1P1  16      +   16318662   16318711   \n",
       "8      Liver cancer           POLD1  19      +   50416152   50416201   \n",
       "9      Liver cancer           STRA6  15      -   74194956   74195005   \n",
       "\n",
       "         p_value     logFC        GSE      Treat_method  genome_pos  \n",
       "0   9.376180e-03 -3.135494  GSE119963   PARPi resistant    54939876  \n",
       "1   1.473765e-02 -2.251292  GSE119963   PARPi resistant    19111209  \n",
       "2   1.740151e-02 -2.140066  GSE119963   PARPi resistant    56963024  \n",
       "3   4.911331e-03 -1.853891  GSE119963   PARPi resistant   106053855  \n",
       "4   2.568635e-03 -1.739703  GSE119963   PARPi resistant    18210373  \n",
       "..           ...       ...        ...               ...         ...  \n",
       "5   5.820000e-09  3.157000  GSE102620  METTL3 knockdown    18379207  \n",
       "6   1.400000e-09  3.020425  GSE102620  METTL3 knockdown    15136898  \n",
       "7   2.370000e-07  2.890372  GSE102620  METTL3 knockdown    16318686  \n",
       "8   1.100000e-07  2.708050  GSE102620  METTL3 knockdown    50416176  \n",
       "9   7.120000e-06  2.442347  GSE102620  METTL3 knockdown    74194980  \n",
       "\n",
       "[226341 rows x 11 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Normalize chromosome format\n",
    "m6add_df['chr'] = m6add_df['chr'].astype(str).str.lower().str.replace('chr', '')\n",
    "\n",
    "# Use midpoint of start-end range in GSE as approximate genome position\n",
    "m6add_df['genome_pos'] = ((m6add_df['start'] + m6add_df['end']) / 2).astype(int)\n",
    "\n",
    "m6add_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "25920b43-5288-4f59-a628-8fa31c6789fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fuzzy_merge(m6anet_df, m6add_df, window=3):\n",
    "    matches = []\n",
    "    for idx, row in m6anet_df.iterrows():\n",
    "        chr_match = m6add_df['chr'] == row['chromosome']\n",
    "        pos_match = (m6add_df['genome_pos'] >= row['genome_pos'] - window) & \\\n",
    "                    (m6add_df['genome_pos'] <= row['genome_pos'] + window)\n",
    "        match_df = m6add_df[chr_match & pos_match]\n",
    "        for _, match_row in match_df.iterrows():\n",
    "            combined = row.to_dict()\n",
    "            combined.update({f'm6add_{col}': match_row[col] for col in m6add_df.columns})\n",
    "            matches.append(combined)\n",
    "    return pd.DataFrame(matches)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f6c7bc88-f5d3-42ed-9ab0-c3a070163c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run fuzzy matching\n",
    "matched_df = fuzzy_merge(m6anet, m6add_df, window=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8baf4cdd-3527-45b8-ae8c-e7bb8c4e2116",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Fuzzy matched 4 entries between m6Anet and M6aDD (±3nt).\n",
      "     transcript_id  gene_name m6add_Symbol_id  genome_pos chromosome\n",
      "0  ENST00000379198    B3GALT6         B3GALT6     1234848          1\n",
      "1  ENST00000379198    B3GALT6         B3GALT6     1234815          1\n",
      "2  ENST00000478938       SDF4            SDF4     1218635          1\n",
      "3  ENST00000506640  LINC00115   RP11-206L10.2      778607          1\n"
     ]
    }
   ],
   "source": [
    "# STEP 4: Report and Preview\n",
    "print(f\"✅ Fuzzy matched {len(matched_df)} entries between m6Anet and M6aDD (±3nt).\")\n",
    "print(matched_df[['transcript_id', 'gene_name', 'm6add_Symbol_id', 'genome_pos', 'chromosome']].head())\n",
    "\n",
    "# STEP 5: Save to file\n",
    "matched_df.to_csv(\"fuzzy_matched_m6anet_m6add.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11a606da-85c7-4129-8318-e6db7d20e13e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
